tests/MongoDB.Analyzer.Tests/Infrastructure/TestCasesRunner.cs

```diff
@@ -58,35 +61,51 @@ public static async Task<DiagnosticTestCaseResult> RunTestCase(DiagnosticTestCas
 
     private static async Task<IDictionary<string, DiagnosticTestCaseResult>> ExecuteTestCases(TestsBundleKey testsBundleKey)
     {
+        var isPoco2JsonTest = testsBundleKey.TestFileName.Contains(s_poco2JsonTestPath);
+
         var diagnostics = await DiagnosticsAnalyzer.Analyze(
             testsBundleKey.TestFileName,
             testsBundleKey.DriverVersion,
             testsBundleKey.LinqVersion);
 
         var diagnosticsAndMethodNodes = diagnostics
-            .Where(d => DiagnosticRulesConstants.AllRules.Contains(d.Descriptor.Id))
-            .Select(d =>
-                (Diagnostic: d,
-                 MethodNode: d.Location.SourceTree.GetRoot()
-                 .FindNode(d.Location.SourceSpan)
-                 .Ancestors()
-                 .OfType<MethodDeclarationSyntax>()
-                 .FirstOrDefault()))
-            .Where(d => d.MethodNode != null)
-            .ToArray();
+                                            .Where(d => DiagnosticRulesConstants.AllRules.Contains(d.Descriptor.Id))
+                                            .Select(d =>
+                                                (Diagnostic: d,
+                                                 MethodNode: FindMethodNode(d, isPoco2JsonTest)))
+                                            .Where(d => d.MethodNode != null)
+                                            .ToArray();
 
         var result = diagnosticsAndMethodNodes
-            .GroupBy(pair => pair.MethodNode.Identifier.Text)
-            .ToDictionary(
-                group => group.Key,
-                group => new DiagnosticTestCaseResult(
-                    group.Key,
-                    GetMethodLocation(group.FirstOrDefault().MethodNode),
-                    group.Select(pair => pair.Diagnostic).ToArray()));
+                    .GroupBy(pair => pair.MethodNode.Identifier.Text)
```

Comment:
Revert indentation changes and adjust elsewhere.

Reply:
Done